DSharpPlus/Net/Gateway/Compression/Zstd/ZstdInterop.Bindings.cs

```diff
@@ -0,0 +1,47 @@
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using ZstdErrorCodeConvertible = nuint;
+
+namespace DSharpPlus.Net.Gateway.Compression.Zstd;
+
+internal unsafe partial struct ZstdInterop
+{
+    // ZstdInterop.Bindings is a nested type to lazily load zstd. the native load is done by the static constructor,
+    // which will not be executed unless this code actually gets used. since we cannot rely on zstd being present at all
+    // times, it is imperative this remains a nested type.
+    private static partial class Bindings
+    {
+        [LibraryImport("libzstd")]
+        [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
```

Comment:
Isn't `Cdecl` the default calling convention anyways? Not sure if this is needed or redundant

Reply:
only on unix, on windows the default cconv is `stdcall`